R/load_package.r

```diff
@@ -1,15 +1,100 @@
 #' Load a package and install it if necessary
 #' 
 #' @name load_package
-#' @param name Name of package
+#' @param name character. Name of package.
+#' @param verbose logical. Whether or not to announce each installation.
 #' @examples
 #' \dontrun{
-#' load_package('glmnet')
+#' load_package("glmnet")
+#' load_package("robertzk/Ramd")
+#' load_package("robertzk/Ramd@@v0.3")   # Can load from versions
+#' load_package("robertzk/Ramd@@0.3")
+#' load_package("robertzk/Ramd@@fbe1aa0e36df289b27881d077635352e6debdbc1")  # Can load from refs
+#' load_package(list("FeiYeYe/xgboost", subdir = "R-package"))  # Can load from subdirectories
 #' }
-load_package <- function(name) {
-  if (!require(name, character.only = TRUE)) {
-    install.packages(name, dependencies = TRUE)
-    if (!require(name, character.only = TRUE))
-      stop(paste('Package', name, 'not found'))
+load_package <- function(name, verbose = FALSE) {
+  metadata <- name[-1]  # For tracking things like subdir
+  name <- name[[1]]
+
+  if (package_is_installed(name)) {
+    if (isTRUE(verbose)) { message(name, " already installed.") }
+    return(TRUE)
   }
+
+  handle_version_mismatches(name, verbose)
+
+  if (is_github_package(name)) {
+    remote <- "GitHub"
+    install_from_github(name, metadata, remote, verbose)
```

Comment:
Can you remove the `remote` parameter? That should be hardcoded into the `install_from_github` method.